### PR TITLE
Partially addresses #1908

### PR DESCRIPTION
**Pull request by wa4557**
This allows changing the tray icon, like it was before, by changing icons in ~/.TelegramDesktop/tdata/ticons. Cf. #1908

### CODE REVIEW

**Comment by auchri** [comment]
Hi,
thanks for this pull request :)

Please read our [contributing policy](https://github.com/telegramdesktop/tdesktop/blob/master/.github/CONTRIBUTING.md). You'll need to make a pull request with the "Signed-off-by:" signature being the last line of your commit message, like it is described in [sign your work](https://github.com/telegramdesktop/tdesktop/blob/master/.github/CONTRIBUTING.md#sign-your-work) section. That will grant your work into the public domain.

(See travis build)

**Comment by john-preston** [comment]
Unfortunately I've changed the way (the place where) the libraries are loaded, new version is available in the dev branch -- so this pull request won't merge. But I think I can add this by myself.

**Comment by wa4557** [comment]
Fine with me, as long as the feature gets added :)

File: Telegram/SourceFiles/pspecific_linux.cpp [hunk 1]
```diff
@@ -641,18 +653,35 @@ void PsMainWindow::psUpdateCounter() {
...
-            ps_gtk_status_icon_set_from_pixbuf(_trayIcon, _trayPixbuf);
+            QFileInfo f(_trayIconImageFile());
+            if (f.exists()) {
+                QByteArray path = QFile::encodeName(f.absoluteFilePath());
+                _trayIcon = ps_gtk_status_icon_new_from_file(path.constData());
+                }
+            else{
+                loadPixbuf(_trayIconImageGen());
+                _trayIcon = ps_gtk_status_icon_new_from_pixbuf(_trayPixbuf);
+            }
```
Comment (john-preston) [inline Telegram/SourceFiles/pspecific_linux.cpp]: You've replaced a ..set_from_pixbuf(_trayIcon) call with a _trayIcon = ...new_from_...(), which means you create each time a new icon, just forgetting a pointer to the old one -- this should cause memory leaks and other bugs (for example, you don't connect callbacks, like it is done on 792 (old) / 820 (new) line. I think you need not only ..._new_from_file(), but both that one and ..._set_from_file().

**Comment by john-preston** [comment]
You've replaced a ..set_from_pixbuf(\_trayIcon) call with a \_trayIcon = ...new_from_...(), which means you create each time a new icon, just forgetting a pointer to the old one -- this should cause memory leaks and other bugs (for example, you don't connect callbacks, like it is done on 792 (old) / 820 (new) line. I think you need not only ..._new_from_file(), but both that one and ..._set_from_file().

**Comment by john-preston** [comment]
I've added such code to the dev branch in the new files.

**Pull request closed without merging**